### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by nanandn** [storybookjs/storybook#10803]
**Describe the bug**
After upgrading to v6.0.0-beta.7 (from 5.3.18), when I run storybook I see an error.

```
hooks.js:232 Uncaught TypeError: Cannot read property 'name' of undefined
    at hooks.js:232
    at decorators.js:35
    at hooks.js:275
    at storyFn (story_store.js:486)
    at renderWithHooks (react-dom.development.js:2546)
    at mountIndeterminateComponent (react-dom.development.js:2831)
    at beginWork (react-dom.development.js:3049)
    at HTMLUnknownElement.callCallback (react-dom.development.js:70)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:90)
    at invokeGuardedCallback (react-dom.development.js:105)
```

**Comment by nanandn** [comment]
Found the problem. It is the @storybook/addon-a11y addon. Looks like 5.x needed addDecorator(withAlly) and 6.x does not.

**Comment by shilman** [comment]
https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-witha11y-decorator

**Comment by shilman** [comment]
@tooppaaa i'm wondering whether we should leave the `withA11y` decorator in, but make it throw an error that points to the migration docs?

**Comment by tooppaaa** [comment]
That would propably ease the translation. I'll do it.
Is there any "tag" to put on the code like "Delete in 7.x" ?

**Comment by shilman** [comment]
@tooppaaa we use `util-deprecate` and then strip them all out on the next major release

**Comment by shilman** [comment]
Yo-ho-ho!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.9 containing PR #10814 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.

Closing this issue. Please re-open if you think there's still more to do.

**Comment by deniseadissi** [comment]
Hi there!
I am facing this very same issue after upgrading to 6.1.20.
Any idea where this might come from?

**Comment by shilman** [comment]
@deniseadissi have you upgraded all your packages using `npx sb upgrade`?

**Comment by deniseadissi** [comment]
@shilman , yes I did, but let me give you a detailed explanation:
I migrated my code to a new repository. `package-lock.json` and `node_modules` directory were not versioned, so they did not get pushed to the new one. 
Then I ran `npm install` and got this error: 

`npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: ui-library@1.0.0
npm ERR! Found: @storybook/addons@6.1.20
npm ERR! node_modules/@storybook/addons
npm ERR!   dev @storybook/addons@"^6.0.26" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @storybook/addons@"3.1.2" from storybook-addon-versions@0.4.3
npm ERR! node_modules/storybook-addon-versions
npm ERR!   dev storybook-addon-versions@"^0.4.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.`

After that, I ran `npm install --legacy-peer-deps`, as suggested by the console. That seemed to work, `npm run storybook` worked fine, but then `npm run chromatic` threw build errors, and on the terminal it suggested to run `npm run build-storybook`. That also failed (ERR! Addon value should end in /register OR it should be a valid preset https://storybook.js.org/docs/presets/introduction/
ERR! @storybook/addon-a11y), and that URL throws a 404, but as  I saw  on the first line `info @storybook/react v6.1.20`, i thought I should run `npx sb upgrade` to make al the other dependencies update accordingly.
That did not show any errors, so I went back and run `npm run storybook`, showing no errors on the terminal, but on the browser the story does not render and on its console the said error appears.

**Comment by shilman** [comment]
What version of `@storybook/addon-a11y` are you running?
